src/main.rs

```diff
@@ -576,6 +492,34 @@ async fn start(
 	Ok(status)
 }
 
+/// Spawn desktop components concurrently with the process manager.
+async fn spawn_components(
+	process_manager: &ProcessManager,
+	env_vars: &[(String, String)],
+	socket_tx: &UnboundedSender<Vec<UnixStream>>,
+) {
+	let components = [
+		"cosmic-app-library",
+		"cosmic-bg",
+		"cosmic-files-applet",
+		"cosmic-greeter",
+		"cosmic-idle",
+		"cosmic-launcher",
+		"cosmic-osd",
+		"cosmic-workspaces",
+	];
+
+	components
+		.into_iter()
+		.map(|name| async move {
+			let span = info_span!(parent: None, "{}", name);
```

Comment:
Does creating an `info_span!` in a loop like this work properly?

https://docs.rs/tracing/0.1.41/tracing/struct.Span.html takes `&'static Metadata<'static>` , and `macro_rules! span` in https://github.com/tokio-rs/tracing/blob/main/tracing/src/macros.rs includes a definition of a `static`.